File: 2021/6/main.jl
using DataStructures

function solve(n)
    fish = counter(parse.(Int, split(read("input.txt", String), ',')))
    
    for _ in 1:n
        x = fish[0]
        for i in 1:9
            fish[i-1] = fish[i]
        end
        fish[6] += x
        fish[8] = x
    end
    println(sum(fish))
end
solve(80)
solve(256)

File: 2021/6/main.py
fish = [int(i) for i in open("input.txt").read().split(",")]
# def simulate():
#     for i in range(len(fish)):
#         if fish[i] == 0:
#             fish[i] = 6
#             fish.append(8)
#         else:
#             fish[i] -= 1
# for i in range(256):
#     print(i)
#     simulate()
# print(len(fish))

def solve(days):
    d = [0]*9
    for i in fish:
        d[i] += 1
    for _ in range(days):
        x = d[0]
        for i in range(1, 9):
            d[i-1] = d[i]
        d[6] += x
        d[8] = x
    return sum(d)
print(solve(80))
print(solve(256))